File: Problem-88-Merge_Sorted_Array/my_sol.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        
        int i = 0; 
        int j = 0;
        vector<int> arr(m+n);
        int x = 0;

        while (i < m || j < n) {
            if (i < m && (j >= n || nums1[i] < nums2[j])) {
                arr[x++] = nums1[i++];
            } else {
                arr[x++] = nums2[j++];
            }
        }

        for(int k = 0; k < m+n; k++) {
            nums1[k] = arr[k];
        }
    }
};


//time complexity O(m+n)
//Space complexity O(m+n)

File: Problem-1534-Count_Good_Triplets/my_sol.py
class Solution:
    def countGoodTriplets(self, arr: List[int], a: int, b: int, c: int) -> int:
        
        tri = 0
        for i in range(len(arr)):
            for j in range(i+1,len(arr)):
                if abs(arr[i] - arr[j]) > a:
                        continue
                for k in range(j+1,len(arr)):
                    
                    if abs(arr[j] - arr[k]) > b:
                        continue
                    if abs(arr[i] - arr[k]) > c:
                        continue
                    
                    tri += 1

        return tri

                    

                

#time complexity: O(n^3) worst case, else O(n^2)
#space complexity: O(1)